clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -1135,21 +1147,19 @@ static void ValidateMultipleRegisterAnnotations(Sema &S, Decl *TheDecl,
 
       RegisterType otherRegType = getRegisterType(attr->getSlot());
       if (RegisterTypesDetected[static_cast<int>(otherRegType)]) {
-        if (PreviousConflicts[TheDecl].count(otherRegType))
```

Comment:
PreviousConflicts is no longer in use as a variable. You could remove it. But it seems unnecessary that you also handle duplicate RBAs in a separate part of the code. Wouldn't it be possible to check here if the decl is a UDT, and load in the relevant data into RBA (the resource field decl) if it is a UDT? 
If not, it might be useful to add a comment above the function stating: "This function validates multiple RBAs for non-udt resource decls, UDT resource decls are validated for duplicate RBAs in `ContainsResourceForRegisterType`"

Reply:
I have removed the unused variable. and the ResourceField is gone as well. I think the duplicate errors are reduced by simply not creating invalid binding attributes, but please let me know if you think otherwise. We should not be emitting duplicate errors for the same binding, but if there are multiple bindings with the same issue we should report them all.